autopilot/autopilot.go

```diff
@@ -97,7 +97,7 @@ type Autopilot struct {
 	workers *workerPool
 
 	mu    sync.Mutex
-	state state
+	state *state
```

Comment:
What's the reason for this change? Doesn't seem like you ever check for `nil` anyway?